starlette/middleware/trustedhost.py

```diff
@@ -53,7 +53,7 @@ async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
             if found_www_redirect and self.www_redirect:
                 url = URL(scope=scope)
                 redirect_url = url.replace(netloc="www." + url.netloc)
-                response = RedirectResponse(url=str(redirect_url))  # type: Response
+                response: Response = RedirectResponse(url=str(redirect_url))
```

Comment:
Is it needed here?

Reply:
Yes, or alternatively:
```python
            response: Response
            if found_www_redirect and self.www_redirect:
                url = URL(scope=scope)
                redirect_url = url.replace(netloc="www." + url.netloc)
                response = RedirectResponse(url=str(redirect_url))
            else:
                response = PlainTextResponse("Invalid host header", status_code=400)
```

Reply:
Without this annotation:
```
+ mypy starlette tests
starlette/middleware/trustedhost.py:58: error: Incompatible types in assignment (expression has type "PlainTextResponse", variable has type "RedirectResponse")
```

Reply:
I usually do what you suggested on the comment (I think it makes more sense). But the current way it's fine.

Reply:
I made the change because I agree it makes more sense.